lib/asciidoctor/section.rb

```diff
@@ -209,6 +209,7 @@ def self.generate_id title, document
       gen_id = gen_id.slice 1, gen_id.length if pre.empty? && (gen_id.start_with? sep)
     end
     if document.catalog[:refs].key? gen_id
+      logger.warn %(auto-generated header id conflict: #{gen_id})
       ids = document.catalog[:refs]
       cnt = Compliance.unique_id_start_index
       cnt += 1 while ids[candidate_id = %(#{gen_id}#{sep}#{cnt})]
```

Comment:
Put the message here instead as:

```
logger.info %(auto-generated ID already in use: #{gen_id}; deduplicated as #{candidate_id}) if logger.info?
```

Also, we'll want to include line information. That may require passing the reader as an additional argument. I'll have to explore that to see if it's feasible.